Missing Initializer Warning in Cuda.hpp

`alpaka/core/Cuda.hpp` throws the following warnings with PIConGPU.

Software environment:
- gcc/4.9.2 (host)
- nvcc/8.0.44 (splitting, device)
- cmake/3.7.2

```
picsrc/thirdParty/alpaka/include/alpaka/core/Cuda.hpp: In instantiation of ‘void alpaka::cuda::detail::cudaRtCheckIgnore(const cudaError_t&, const char*, const char*, const int&, TErrors&& ...) [with TErrors = {}; <template-parameter-1-2> = void; cudaError_t = cudaError]’:
picsrc/thirdParty/alpaka/include/alpaka/dev/DevCudaRt.hpp:125:510:   required from here
picsrc/thirdParty/alpaka/include/alpaka/core/Cuda.hpp:122:126: warning: missing initializer for member ‘std::array<cudaError, 0ul>::_M_elems’ [-Wmissing-field-initializers]
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
                                                                                                                              ^
picsrc/thirdParty/alpaka/include/alpaka/core/Cuda.hpp: In instantiation of ‘void alpaka::cuda::detail::cudaRtCheckIgnore(const cudaError_t&, const char*, const char*, const int&, TErrors&& ...) [with TErrors = {}; <template-parameter-1-2> = void; cudaError_t = cudaError]’:
picsrc/thirdParty/alpaka/include/alpaka/dev/DevCudaRt.hpp:125:510:   required from here
picsrc/thirdParty/alpaka/include/alpaka/core/Cuda.hpp:122:126: warning: missing initializer for member ‘std::array<cudaError, 0ul>::_M_elems’ [-Wmissing-field-initializers]
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
```

If I enable the `missing braces` warning I get straight to the `-Wmissing-field-initializers` warning:
```
alpaka/include/alpaka/core/Cuda.hpp: In instantiation of ‘void alpaka::cuda::detail::cudaRtCheckIgnore(const cudaError_t&, const char*, const char*, const int&, TErrors&& ...) [with TErrors = {}; <template-parameter-1-2> = void; cudaError_t = cudaError]’:
alpaka/include/alpaka/dev/DevCudaRt.hpp:125:510:   required from here
alpaka/include/alpaka/core/Cuda.hpp:127:126: warning: missing initializer for member ‘std::array<cudaError, 0ul>::_M_elems’ [-Wmissing-field-initializers]
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
                                                                                                                              ^
alpaka/include/alpaka/core/Cuda.hpp: In instantiation of ‘void alpaka::cuda::detail::cudaRtCheckIgnore(const cudaError_t&, const char*, const char*, const int&, TErrors&& ...) [with TErrors = {}; <template-parameter-1-2> = void; cudaError_t = cudaError]’:
alpaka/include/alpaka/dev/DevCudaRt.hpp:125:510:   required from here
alpaka/include/alpaka/core/Cuda.hpp:127:126: warning: missing initializer for member ‘std::array<cudaError, 0ul>::_M_elems’ [-Wmissing-field-initializers]
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
```

--- a/include/alpaka/core/Cuda.hpp
+++ b/include/alpaka/core/Cuda.hpp
@@ -114,22 +114,9 @@ namespace alpaka
             {
                 if(error != cudaSuccess)
                 {
-                    // Disable the incorrect warning see: http://stackoverflow.com/questions/13905200/is-it-wise-to-ignore-gcc-clangs-wmissing-braces-warning
-#if BOOST_COMP_CLANG
-    #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wmissing-braces"
-#endif
-#if BOOST_COMP_GNUC
-    #pragma GCC diagnostic push
-    #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
-#endif
-                    std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
-#if BOOST_COMP_CLANG
-    #pragma clang diagnostic pop
-#endif
-#if BOOST_COMP_GNUC
-    #pragma GCC diagnostic pop
-#endif
+                    // https://stackoverflow.com/questions/18792731/can-we-omit-the-double-braces-for-stdarray-in-c14/18792782#18792782
+                    std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{{ignoredErrorCodes...}};
+
                     // If the error code is not one of the ignored ones.
                     if(std::find(aIgnoredErrorCodes.cbegin(), aIgnoredErrorCodes.cend(), error) == aIgnoredErrorCodes.cend())
                     {

--- a/include/alpaka/core/Cuda.hpp
+++ b/include/alpaka/core/Cuda.hpp
@@ -119,9 +119,16 @@ namespace alpaka
     #pragma clang diagnostic push
     #pragma clang diagnostic ignored "-Wmissing-braces"
 #endif
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
 #if BOOST_COMP_CLANG
     #pragma clang diagnostic pop
+#endif
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic pop
 #endif
                     // If the error code is not one of the ignored ones.
                     if(std::find(aIgnoredErrorCodes.cbegin(), aIgnoredErrorCodes.cend(), error) == aIgnoredErrorCodes.cend())

